src/rpdk/core/boto_helpers.py

```diff
@@ -78,7 +83,21 @@ def get_service_endpoint(service, region):
     return "https://" + endpoint_data["hostname"]
 
 
-def get_account(session):
-    sts_client = session.client("sts")
+def get_account(session, temporary_credentials=None):
+    if temporary_credentials:
+        sts_client = session.client(
+            "sts",
+            endpoint_url=get_service_endpoint("sts", session.region_name),
+            region_name=session.region_name,
+            aws_access_key_id=temporary_credentials["accessKeyId"],
+            aws_secret_access_key=temporary_credentials["secretAccessKey"],
+            aws_session_token=temporary_credentials["sessionToken"],
+        )
+    else:
```

Comment:
Why we need to create a sts_client from default credential prodiver?    The `get_temporary_credentials` should cover all the cases.

Reply:
You are right; get_temporary_credentials should cover all the cases. I was incorrectly thinking that the temporary credentials would only be used when a role is specified (which is not the case). Will change

Reply:
You are right; get_temporary_credentials should cover all the cases. I was incorrectly thinking that the temporary credentials would only be used when a role is specified (which is not the case). Will change